### ISSUE TITLE
Retry Batching consumes too many resources and crashes SC

### ISSUE THREAD

**Issue by cjoergensen** [Particular/ServiceControl#501]
## Who's affected

Users of ServiceControl 1.6.0 and up with large numbers of failed messages to retry.

## Symptoms

When retrying a large number of messages, ServiceControl maxes the HDD utilization and then memory until a Raven error occurs. This is caused by an unbounded parallel execution during retry batch creation. 

--
This fix pushes all retry requests into a background queue and processes them one at a time.
If this issue persists then the embedded RavenDB may require some tweaking. Add the following line to the ServiceControl config file:

`<add key="Raven/Esent/MaxVerPages" value="1024" />`

The value is the size of the "version store" used for open transactions in megabytes. The default value is 512MB. You can increase this to 2048 or 4096 if necessary.

**Comment by cjoergensen** [comment]
Im sorry, the proper version is 1.6.1

**Comment by wolfbyte** [comment]
@cjoergensen - it sounds like with that many messages that the embedded Raven instance may need some tweaking. 

Can you provide some information about the machine that ServiceControl is running on? 

Which operating system?
32 or 64 bit?
RAM?
What else is running on the machine?

If you are able to send me a backup of your database I can try a few tweaks to see if I can help you resolve it. mike.minutillo-at-particular.net

**Comment by mikeminutillo** [comment]
Original Issue Details

Hi

We are retrying a lot of failed message (200.000+) and ServiceControl keeps on stopping.

When investigation the Windows Event log I found this entry just before the service stops:
```
------------------------------------------------------------------------
Raven (13472) 1-szk08-E:\RavenDB\ServiceControl.1.4.2\Data: The version store for this instance (0) has reached its maximum size of 511Mb. It is likely that a long-running transaction is preventing cleanup of the version store and causing it to build up in size. Updates will be rejected until the long-running transaction has been completely committed or rolled back. 
Possible long-running transaction: 
	SessionId: 0x000000387F184640 
	Session-context: 0x00000000 
	Session-context ThreadId: 0x0000000000004488 
	Cleanup: 1 
	Session-trace: 
57573@20:04:35
------------------------------------------------------------------------
```
The ServiceControl log file shows these entries:

NServiceBus.Unicast.Transport.TransportReceiver|Failed to process message
Microsoft.Isam.Esent.Interop.EsentVersionStoreOutOfMemoryException: Version store out of memory (cleanup already attempted)

**Comment by mikeminutillo** [comment]
@cjoergensen I have updated the details of this issue so our release notes generation will pick it up. I put the details of the original issue in a comment above so they are not lost. 

This should be fixed as a part of the 1.7 release. If this release does not solve the problem for you feel free to re-open this issue or create a new issue.
